snapshot-labs/snapshot-sidekick
# feat: add support for snapshot's hub api key

## 🧿 Current issues / What's wrong ?

Sidekick server is currently rate limited, preventing the CSV votes caching feature to be  deployed. The limit of 60 requests/20seconds is not enough, for a server centralizing all votes requests for snapshot UI, and all the requests to the hub are constantly hit with a 429 error.

## 💊 Fixes / Solution

FIx #19 

Keycard has been developed to cater to this use case. Using API key will solve the throttling issue.

## 🚧 Changes

- Send the `x-api-key` header, with an API key

## 🛠️ Tests

- Set `KEYCARD_API_KEY` in `.env`
- Call any endpoint requiring a hub request (`curl -X POST http://localhost:3000/api/votes/0x223db03c09d41e0280bbe53fca84cdf44547c046e1fcc3272aea6c39e4278291`)
- It should create a cache file (look at the console output)
